test/GPv2TradeExecution.test.ts

```diff
@@ -85,10 +81,10 @@ describe("GPv2TradeExecution", () => {
           }),
           recipient.address,
         ),
-      ).to.be.revertedWith("call to non-contract");
+      ).not.to.be.reverted;
     });
 
-    it("should revert when mistakenly trying to sell ETH using the marker buy Ether address", async () => {
+    it("does not revert when mistakenly trying to sell ETH using the marker buy Ether address", async () => {
```

Comment:
So the reason I duplicated this test, is that I think that setting `buyToken` to ETH should revert (while the above case I thought could change based on #264). Personally, I think we should not regress on this behaviour and maybe add a specific check in the `GPv2TransferManager`.

Reply:
I'd be ok with ignoring this edge case. It doesn't really harm the user, since if the order is actually executed then the user would get free tokens, basically. In any case, I added an extra check that forbids transfers involving the marker ETH address: this extra check costs about 40 gas/trade, so not too much.

Reply:
> I'd be ok with ignoring this edge case. It doesn't really harm the user, since if the order is actually executed then the user would get free tokens, basically.

Hmm... You're right. Maybe this check is just not needed after all and can be removed (with the unit test as well).